File: 0066-plus-one/0066-plus-one.go
func plusOne(digits []int) []int {
    for i := len(digits)-1; i > -1; i-- {
        if digits[i] < 9 {
            digits[i]++
            return digits
        }
        digits[i] = 0
    }
    
    return append([]int{1}, digits...)
}

File: 0066-plus-one/0066-plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    if (digits.length === 0) {
        return [1]
    }
    
    if (digits[digits.length-1] + 1 > 9) {
        let ret = plusOne(digits.slice(0, -1))
        ret.push(digits[digits.length-1]-9)
        return ret
    }
    
    let ret = digits.slice(0, -1)
    ret.push(digits[digits.length-1]+1)
    return ret
};